### PR TITLE
Take upstream impls and segregate modifications

### PR DESCRIPTION
**Pull request by ureeves**
The upstream implementations are taken verbatim, and our own modifications are added in a more segregated manner, for ease of identification.

Resolves: #115, #110

### CODE REVIEW

**Review by moCello — APPROVED** [review]
LGTM, some notes but they are mostly for possible future investigations

File: src/lib.rs [hunk 1]
```diff
@@ -34,281 +33,266 @@
...
+pub use fr::Fr;
 
-pub(crate) use fr::Fr;
+/// Represents an element of the base field $\mathbb{F}_q$ of the Jubjub
+/// elliptic curve construction.
+pub type Base = Fq;
 
-/// A better name than Fr.
+/// Represents an element of the scalar field $\mathbb{F}_r$ of the Jubjub
+/// elliptic curve construction.
```
[the remark this answers is not in the document]
Comment (ureeves) [reply src/lib.rs:75]: These are the names upstream uses. We decided to name them `BlsScalar` and `JubJubScalar`, which are now also type aliases in `mod dusk` underneath.

File: src/lib.rs [hunk 2]
```diff
@@ -1,18 +1,17 @@
 //! This crate provides an implementation of the **Jubjub** elliptic curve and
-//! its associated field arithmetic.
-//! See [`README.md`](https://github.com/zkcrypto/jubjub/blob/master/README.md)
-// for more details about Jubjub.
+//! its associated field arithmetic. See [`README.md`](https://github.com/zkcrypto/jubjub/blob/master/README.md) for more details about Jubjub.
```
[the remark this answers is not in the document]
Comment (ureeves) [reply src/lib.rs:2]: Will do

File: src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (moCello) [reply src/lib.rs:75]: Yeah, I like that we go with the version zkcrypto is using and the inconsistency of exposing the same type under two names seems to be in their upstream version: jubjub-scalar is exposed as both `Scalar` and `Fr` and bls-scalar is exposed as both `Base` and `Fq`.

**Pull request merged**